### DISCUSSION TITLE
Custom endpoint error

### DISCUSSION THREAD

**Discussion by ghost** [directus/directus#4009] [Archive]
Hi i'm trying to make a custom endpoint using the example with adaptation 
`
module.exports = function registerEndpoint(router, { services, exceptions }) {
	const { ItemsService } = services;
	const { ServiceUnavailableException } = exceptions;

	router.get('/', (req, res) => {
		const recipeService = new ItemsService('test_data', { schema: req.schema });
		recipeService
		.readByQuery({sort:'value',fields:'*'})
		.then((results) => res.json(results))
		.catch((error) => {
			throw new ServiceUnavailableException(error.message);
		});
	});
};
`

but I get an error when I call my endpoint

`(node:7776) UnhandledPromiseRejectionWarning: Error: fields.splice is not a function
    at C:\Users\efernandez\Desktop\data\test-data\extensions\endpoints\users\index.js:11:10
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
(node:7776) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
(node:7776) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
`

can you tell me what I did wrong ? I'm using the v9
Thanks !

**Comment by rijkvanzanten** [comment]
The type of `fields` in the `readByQuery` is an array of strings:

`.readByQuery({sort:'value',fields:'*'})`

should be

`.readByQuery({ sort:'value', fields: ['*'] })`

👍🏻

**Reply by ghost** [reply]
Ok thanks ! I see you change the doc :) 
Thanks again !

**Reply by rijkvanzanten** [reply]
Technically @pzehle did 😄 https://github.com/directus/directus/pull/4012
